File: robochecker/Lab3/fileReverse.c
/*
*  ID: 1574354
*  Nathan Baledio
*  fileReverse.c
*  Reads input file and prints each word, reverse on a separate line of
*  the output file.
*/

#include<stdio.h>
#include<stdlib.h>

void stringReverse(char* s);

void stringReverse(char* s){
	char reverse[256];
	memset(reverse, 0, 256);
	int i;
	int j = 0;
	int k;
	for(i = strlen(s) - 1; i >= 0; i--){
		reverse[j] = s[i];
		j++;
	}
	for(k = 0; k <= strlen(reverse) - 1; k++){
		s[k] = reverse[k];
	}
}

int main(int argc, char* argv[]){
   FILE* in;  /* file handle for input */  
   FILE* out; /* file handle for output */
   char word[256]; /* char array to store words from input file */

   /* check command line for correct number of arguments */
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   /* open input file for reading */
   in = fopen(argv[1], "r");
   if( in==NULL ){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   /* open output file for writing */
   out = fopen(argv[2], "w");
   if( out==NULL ){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   /* read words from input file, print on separate lines to output file*/
   while( fscanf(in, " %s", word) != EOF ){
      stringReverse(word);
     fprintf(out, "%s\n", word);
   } 

   /* close input and output files */
   fclose(in);
   fclose(out);

   return(EXIT_SUCCESS);
}

File: Lab2/FileReverse.java
//Nathan Baledio
//ID 1574354
import java.io.*;
import java.util.Scanner;
class FileReverse{
		 public static String stringReverse(String s) {
			 String string = s;
			 String reverse=""; 
			 for(int i=string.length()-1;i>=0;i--){
				 reverse+=string.charAt(i);
			 }
			 return reverse;
		 }
   public static void main(String[] args) throws IOException{

      int lineNumber = 0;

      // check number of command line arguments is at least 2
      if(args.length < 2){
         System.out.println("Usage: java –jar FileTokens.jar <input file> <output file>");
         System.exit(1);
      }

      // open files
      Scanner in = new Scanner(new File(args[0]));
      PrintWriter out = new PrintWriter(new FileWriter(args[1]));

      // read lines from in, extract and print tokens from each line
      while( in.hasNextLine() ){
         lineNumber++;

         // trim leading and trailing spaces, then add one trailing space so 
         // split works on blank lines
         String line = in.nextLine().trim() + " "; 

         // split line around white space 
         String[] token = line.split("\\s+");  

         // print out tokens       
         int n = token.length;
         for(int i=0; i<n; i++){
            token[i]=stringReverse(token[i]);
         }
		 for(int i=0; i<n; i++){		 
            out.println(token[i]);
         }
		
		 }
		// close files
		in.close();
		out.close();
      }


   }



